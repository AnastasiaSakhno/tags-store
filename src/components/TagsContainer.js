import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import actions from '../actions'
import TagList from './TagList'
import TagForm from './TagForm'


class TagsContainer extends Component {
  static propTypes = {
    loadTags: PropTypes.func.isRequired,
    addTag: PropTypes.func.isRequired,
    tags: PropTypes.array.isRequired
  }

  componentDidMount() {
    this.props.loadTags()
  }

  render() {
    const tags = this.props.tags.filter( (tag) => {
      return tag.linkId === this.props.linkId
    })

    return (
      <div className="tags-box">
        <TagForm onTagSubmit={ this.props.addTag } linkId={ this.props.linkId }/>
        <TagList tags={ tags } onDestroy={ this.props.removeTag }/>
      </div>
    )
  }
}

const mapStateToProps = (state) => ({
  tags: state.tags
})

const mapDispatchToProps = (dispatch) => ({
  loadTags: () => {
    dispatch(actions.tags.load())
  },
  addTag: (tag) => {
    dispatch(actions.tags.add(tag))
  },
  removeTag: (tag) => {
    dispatch(actions.tags.remove(tag))
  }
})

export default connect(mapStateToProps, mapDispatchToProps)(TagsContainer)
