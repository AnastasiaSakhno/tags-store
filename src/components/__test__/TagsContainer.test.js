import React from 'react'
import renderer from 'react-test-renderer'
import TagsContainer from '../TagsContainer'
import uniqId from '../../utils/uniq-id'

const linkId = uniqId()
const link = {
  id: linkId,
  name: 'test link',
  url: 'https://test.link.ua',
  tags: [
    { linkId: linkId, name: 'one' },
    { linkId: linkId, name: 'two' }
  ],
  archive: false
}
const addTag = () => {}
const removeTag = () => {}

it('renders correctly', () => {
  const tree = renderer
    .create(<TagsContainer linkId={ link.id } tags={ link.tags } addTag={ addTag } removeTag= { removeTag }/>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
