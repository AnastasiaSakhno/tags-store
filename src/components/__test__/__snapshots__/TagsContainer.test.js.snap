// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="tags-box"
>
  <form
    className="tag-form"
    onSubmit={[Function]}
  >
    <input
      placeholder="Tag name"
      type="text"
    />
    <input
      type="submit"
      value="Add tag"
    />
  </form>
  <div
    className="tag-list"
  >
    <span
      className="tag"
    >
      one
      <span
        className="iconbutton-box"
        onClick={[Function]}
      >
        <i
          className="fa fa-remove "
        />
      </span>
    </span>
    <span
      className="tag"
    >
      two
      <span
        className="iconbutton-box"
        onClick={[Function]}
      >
        <i
          className="fa fa-remove "
        />
      </span>
    </span>
  </div>
</div>
`;
